packages/fork-choice/src/forkChoice/forkChoice.ts

```diff
@@ -665,8 +665,37 @@ export class ForkChoice implements IForkChoice {
     return this.protoArray.isDescendant(ancestorRoot, descendantRoot);
   }
 
+  /**
+   * All indices in votes are relative to proto array so always keep it up to date
+   */
   prune(finalizedRoot: RootHex): ProtoBlock[] {
-    return this.protoArray.maybePrune(finalizedRoot);
+    const prunedNodes = this.protoArray.maybePrune(finalizedRoot);
+    const prunedCount = prunedNodes.length;
+    for (const vote of this.votes) {
```

Comment:
If this array if not sparse, it's always faster to use a regular for loop

Reply:
will work on this in another PR, merging to see if it helps the I/O lag issue in `subsribe-all-subnets` mode

Reply:
addressed in https://github.com/ChainSafe/lodestar/pull/5907